go/libraries/doltcore/diff/async_differ.go

```diff
@@ -16,17 +16,41 @@ package diff
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"time"
 
 	"golang.org/x/sync/errgroup"
 
+	"github.com/dolthub/dolt/go/libraries/doltcore/row"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
 	"github.com/dolthub/dolt/go/libraries/utils/async"
 	"github.com/dolthub/dolt/go/store/diff"
 	"github.com/dolthub/dolt/go/store/types"
 )
 
+type RowDiffer interface {
+	// Start starts the RowDiffer.
+	Start(ctx context.Context, from, to types.Map)
+
+	// GetDiffs returns the requested number of diff.Differences, or times out.
+	GetDiffs(numDiffs int, timeout time.Duration) ([]*diff.Difference, bool, error)
+
+	// Close closes the RowDiffer.
+	Close() error
+}
+
+func NewRowDiffer(ctx context.Context, fromSch, toSch schema.Schema, buf int) RowDiffer {
+	ad := NewAsyncDiffer(buf)
+
+	// assumes no PK changes
```

Comment:
Expand on this comment a bit. What happens if one schema is keyless and one not?